components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -522,6 +526,20 @@ private void validateClaimProperties(Map<String, String> claimProperties) throws
                         String.format(ERROR_CODE_CLAIM_PROPERTY_CHAR_LIMIT_EXCEED.getMessage(), property.getKey(),
                                 MAX_CLAIM_PROPERTY_LENGTH));
             }
+            if (StringUtils.equalsIgnoreCase(ClaimConstants.MIN_LENGTH, property.getKey()) ||
+                    StringUtils.equalsIgnoreCase(ClaimConstants.MAX_LENGTH, property.getKey())) {
+                checkMinMaxLimit(property.getKey(), property.getValue());
+            }
+        }
+    }
+
+    private void checkMinMaxLimit(String property, String value) throws ClaimMetadataClientException {
+
+        if (!NumberUtils.isNumber(value) || Integer.parseInt(value) < MIN_CLAIM_PROPERTY_LENGTH_LIMIT ||
```

Comment:
Is the String value always an integer value, Don't we need to consider the cases where we get non-parsable values(non-integer values)?

Reply:
The expected value for these props are numbers. So we should not accept a non numeric value here.